core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -109,6 +108,85 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     assert(!failedTaskSet)
   }
 
+  test("Scheduler does not always schedule tasks on the same workers") {
+    val taskScheduler = setupScheduler()
+    roundrobin(taskScheduler)
+  }
+
+  test("User can specify the roundrobin task assigner") {
+    val taskScheduler = setupScheduler(("spark.scheduler.taskAssigner", "RoUndrObin"))
+    roundrobin(taskScheduler)
+  }
+
+  test("Fallback to roundrobin when the task assigner provided is not valid") {
+    val taskScheduler = setupScheduler("spark.scheduler.taskAssigner" -> "invalid")
+    roundrobin(taskScheduler)
+  }
+
+  test("Scheduler balance the assignment to the worker with more free cores") {
```

Comment:
I think you should structure this so that tests for the same task assigner are packed together. Right now, one has the look into each test case to figure out which policy is being tested.